util/network-devp2p/src/discovery.rs

```diff
@@ -728,7 +743,7 @@ impl<'a> Discovery<'a> {
 		trace!(target: "discovery", "Got {} Neighbours from {:?}", results_count, &from);
 		for r in rlp.at(0)?.iter() {
 			let endpoint = NodeEndpoint::from_rlp(&r)?;
-			if !endpoint.is_valid() {
+			if !endpoint.is_valid_sync_node() {
```

Comment:
Shouldn't it be `is_valid_bootnode` here?

Reply:
`endpoint.is_valid_sync_node == endpoint.is_valid`

Reply:
Well that's kind of my point. Here we are receiving neighbors, and I guess it would be possible that some of those are actually _bootnodes_. I didn't think of that earlier, but here I think you should revisit `let endpoint = NodeEndpoint::from_rlp(&r)?;` and `endpoint.is_valid_sync_node` to take this case into account: we want to add those to our discovery list, without adding them to our syncing list.